### ISSUE TITLE
Better Loss Functions that handle Imbalance

### ISSUE THREAD

**Issue by williamshen-nz** [williamshen-nz/learning-competences#13]
Focal loss, dice loss, weighted cross entropy, etc.

**Comment by williamshen-nz** [comment]
Focal loss isn't great - leave aside for now
